### PR TITLE
[extractor/prankcast] Fix fatal error if guests_json is empty

### PR DESCRIPTION
**Pull request by columndeeply**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

Currently if the guests_json is empty we get a "can only concatenate list" error. This wraps the traverse_obj in a try_call and returns an empty list if it fails. I've added a second test to check this.

```
[~] yt-dlp "https://prankcast.com/phonelosers/showreel/2048-NOT-COOL"
[PrankCast] 2048: Downloading webpage
ERROR: can only concatenate list (not "NoneType") to list
```

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/prankcast.py [hunk 1]
```diff
@@ -4,24 +4,44 @@
...
-            'timestamp': 1661391575,
-            'uploader': 'Devonanustart',
-            'channel_id': 4,
-            'duration': 7918,
-            'cast': ['Devonanustart', 'Phonelosers'],
-            'description': '',
-            'categories': ['prank'],
-            'tags': ['prank call', 'prank'],
-            'upload_date': '20220825'
+    _TESTS = [
```
Comment (Grub4K) [inline yt_dlp/extractor/prankcast.py:7]: Please keep the old format, like `[{`, `}, {` and `}]`.

File: yt_dlp/extractor/prankcast.py [hunk 2]
```diff
@@ -42,7 +62,7 @@ def _real_extract(self, url):
             'uploader': uploader,
             'channel_id': json_info.get('user_id'),
             'duration': try_call(lambda: parse_iso8601(json_info['end_date']) - start_date),
-            'cast': list(filter(None, [uploader] + traverse_obj(guests_json, (..., 'name')))),
+            'cast': list(filter(None, [uploader] + (try_call(lambda: traverse_obj(guests_json, (..., 'name'))) or []))),
```
Comment (Grub4K) [inline yt_dlp/extractor/prankcast.py:65]: `traverse_obj` should never throw and supports a default
```suggestion
            'cast': list(filter(None, [uploader] + traverse_obj(guests_json, (..., 'name'), default=[]))),
```

Comment (pukkandan) [reply yt_dlp/extractor/prankcast.py:65]: Even default isnt needed here. What's wrong with the old code?

File: yt_dlp/extractor/prankcast.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/prankcast.py:7]: either style's fine

**Review by Grub4K — APPROVED** [review]

**Comment by pukkandan** [comment]
This is no longer needed due to #5170, but I'll merge the test separately

**Comment by columndeeply** [comment]
Sure, should I push a commit reversing the changes and leaving just the new test or can you merge just the test directly?

**Comment by pukkandan** [comment]
Don't worry about it. I'll merge directly

**Pull request closed without merging**